src/App.js

```diff
@@ -1,26 +1,87 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import { firebase, db } from './lib/firebase';
+import { useCollection } from 'react-firebase-hooks/firestore';
+
+const App = () => {
+  const [value, loading, error] = useCollection(
+    firebase.firestore().collection('list'),
+    {
+      snapshotListenOptions: { includeMetadataChanges: true },
+    },
+  );
+
+  console.log(firebase);
+  console.log(db);
+
+  const sendItem = () => {
+    firebase.db
```

Comment:
`firebase.db` came back as `undefined` when I ran locally, but following @yenly 's suggestion for line 8, it worked perfectly when altering this:`firebase.db.collection('list')`  to just : `db.collection('list')`

Reply:
Yenly's suggestion causes an error.  I console logged db and it's not an instance of (firebase.firestore()).  It actually comes back as firebase.firestore.Firestore.  I'm assuming that is why we can't refactor our code in this way.

Reply:
| Current | Proposed |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/5881537/114279416-7720cc80-99e9-11eb-8ed2-f809ff2c3295.png) | ![image](https://user-images.githubusercontent.com/5881537/114279394-4e98d280-99e9-11eb-85ca-f03e7c53fe9e.png) |

I was able to get Firebase working successfully with the following. Not all the changes are necessary but do offer some readability benefits:
* In `a)` we are passing an unnamed config `{}` object into the `initializeApp` method. In `1)` we've broken this out into its own object, more clearly explaining what `initializeApp` requires below in `2)`.
* In `a)` and `b)` we are calling methods on `firestore`. While this _does_ work and lines up with some of the documentation that Firebase provides, the `initializeApp` method returns the specific firebase instance. We can make a more clear connection between the configuration and what we're consuming down the road by capturing this in a `const firebaseInstance` here in `2)`. We can see the benefits of this below in the differences between `b)` and `3)`. When asking the question "Where is our `db` coming from?" we can more easily see that our configured instance has a method called `firestore()` on it that returns our firestore database.
* In `c)` we are exporting both a reference to `firestore` as a whole and the configured `db` that we'll be using. Given our story/issue we're solving we only need that configured db, so let's only export what's needed.
We can think of this as an API that we're setting up. A colleague could have a need to use this database in the future and have less familiarity than we do. When we export both items we add ambiguity as to what should be used, which could lead our colleague to importing `firestore` and setting up their own instance of the app and database. By only exporting `db` as shown by `4)` we communicate what should be consumed; our ready-to-go pre-configured database!

Reply:
After the above you can reference only the `db` object and the app should still function as expected:
![image](https://user-images.githubusercontent.com/5881537/114280198-109dad80-99ed-11eb-9279-38fde0516aad.png)

Reply:
Thanks!  We've done a fair amount of reconfiguration work in our firebase file, so tracking down the impolementation errors can be rough.  I appreciate these discriptive instructions with screenshots.  I'll try this locally now.

Reply:
You know what I learned today? 

After making changes and saving them, don't trust the screen that first pops up, especially if it has a bunch of errors. 

Manually refresh it. 

Then start troubleshooting, etc if needed. Oh. My. Goodness.

The changes you suggested, Yenly and Skyler, I have implemented them and the code is working as intended.

And thank you for the very thorough explanation and screenshots, Skyler. What you said made sense.